.github/workflows/integration.yml

```diff
@@ -0,0 +1,66 @@
+name: integration-test
+
+on:
+  push:
+    branches:
+      - master
+      # - <your-branch>    # put your branch name here to test it @ GH Actions
+  pull_request:
+    branches:
+      - master
+      - b0.*
+
+jobs:
+
+  goth-tests:
+    name: Run integration tests
+    runs-on: goth
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8.0'
+
+      - name: Configure poetry
+        uses: Gr1N/setup-poetry@v4
+        with:
+          poetry-version: 1.1.4
+
+      - name: Install dependencies
+        run: |
+          poetry env use python3.8
+          poetry install -E integration-tests
+
+      - name: Log in to GitHub Docker repository
+        run: echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com -u ${{github.actor}} --password-stdin
```

Comment:
do we still need that given that `goth` repo is now public?

Reply:
We still need this for two reasons:
1. Pointing the Docker daemon to GitHub's package registry (by default it only uses hub.docker.com).
2. Basic authentication, I believe this is similar to using GitHub's REST API, i.e. you still need to provide a personal token even when accessing public endpoints, presumably to handle stuff like rate limiting.